### PR TITLE
Added version to the top of the aa model

### LINKED ISSUE #42: Add version marker to generated cost models
To help with possible future Gateway analytics.

### PR DESCRIPTION
**Pull request by carlosvdr**
Issue: https://github.com/semiotic-ai/autoagora/issues/42

### CODE REVIEW

**Review by aasseman — CHANGES_REQUESTED** [review]
- This is not the correct way of checking the version. One of the reasons is that `poetry` is a dev dependency, and is not installed in the container (see the Dockerfile), nor is it required by the pip package. In general, it is unnecessarily risky to run/depend on external binaries.
  See https://docs.python.org/3/library/importlib.metadata.html#distribution-versions
- Smaller note, can we have something like `Generated by AutoAgora {version}`?

File: autoagora/model_builder.py [hunk 1]
```diff
@@ -15,7 +16,8 @@
 async def model_builder(subgraph: str, pgpool: asyncpg.Pool) -> str:
     logs_db = LogsDB(pgpool)
     agora_entries = []
-
+    aa_version = version("wheel")
```
Comment (aasseman) [inline autoagora/model_builder.py:19]:
```suggestion
    aa_version = version("autoagora")
```

**Review by aasseman — CHANGES_REQUESTED** [review]

**Comment by aasseman** [comment]
Looks good. Could you write a unit test to make sure that it works as expected?

File: tests/test_model_builder.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    stddev_time=0.5,
+                )
+            ]
+            model = await model_builder.model_builder(
+                "Qmadj8x9km1YEyKmRnJ6EkC2zpJZFCfTyTZpuqC3j6e1QH", pgpool
+            )
+            pattern = r"\n# Generated by AutoAgora \d+\.\d+\.\d+"
+            # To ensure a version is being obtained
+            assert re.match(pattern, model), f"{model} does not match pattern {pattern}"
+            assert model.startswith("\n# Generated by AutoAgora")
```
Comment (aasseman) [inline tests/test_model_builder.py:50]: Isn't that line redundant with the one above?

**Review by aasseman — CHANGES_REQUESTED** [review]
Looks great! 
Could you remove the newline before `Generated by...`. It just adds an empty newline at the very top of the Agora document.

File: tests/test_model_builder.py (hunk 2 above)
Comment (carlosvdr) [reply tests/test_model_builder.py:50]: Was thinking on the possibility of the version not showing in that specific way, but actually you are right if it doesnt show in that way it is definetly wrong, let me remove it

**Comment by aasseman** [comment]
Perfect! I'm gonna do a bit of the old test in prod and then we're good to go!

**Comment by aasseman** [comment]
Need to merge #63 first

**Comment by aasseman** [comment]
Please rebase again onto latest master, then I'll test again for good measure

**Review by aasseman — APPROVED** [review]
LGTM, tested in prod.

**Pull request merged**